kratos/solving_strategies/strategies/solving_strategy.h

```diff
@@ -25,423 +24,474 @@
     KRATOS_CLASS_POINTER_DEFINITION(SolvingStrategy);
-
-    typedef typename ModelPart::DofType TDofType;
-    typedef typename ModelPart::DofsArrayType DofsArrayType;
-    //typedef Dof<TDataType> TDofType;
-    //typedef PointerVectorSet<TDofType, IdentityFunction<TDofType> > DofsArrayType;
-    /* 		typedef PointerVectorSet<TDofType, IndexedObject> DofsArrayType; */
-    typedef typename DofsArrayType::iterator DofIterator;
-    typedef typename DofsArrayType::const_iterator DofConstantIterator;
-    /*@} */
+    
+    ///@}
+    ///@name Life Cycle
+    ///@{
 
     /** Constructor.
+     * @param rModelPart: The model part to be computed
+     * @param MoveMeshFlag: The flag to set if the mesh is moved or not
      */
 
-    /*@{ */
-
     SolvingStrategy(
-        ModelPart& model_part, bool MoveMeshFlag = false
+        ModelPart& rModelPart, 
+        bool MoveMeshFlag = false
     )
-        : mr_model_part(model_part)
+        : mrModelPart(rModelPart)
     {
         SetMoveMeshFlag(MoveMeshFlag);
     }
-    /*@} */
+    
+    ///@}
 
     /** Destructor.
      */
 
-    /*@{ */
-    virtual ~SolvingStrategy()
-    {
-    }
-    /*@} */
+    ///@{
+    virtual ~SolvingStrategy(){}
+    
+    ///@}
 
-    //*********************************************************************************
     /**OPERATIONS ACCESSIBLE FROM THE INPUT:*/
-    /*@{ */
+    
+    ///@{
 
     /**
-    operation to predict the solution ... if it is not called a trivial predictor is used in which the
-    values of the solution step of interest are assumed equal to the old values
+     * Operation to predict the solution ... if it is not called a trivial predictor is used in which the
+     * values of the solution step of interest are assumed equal to the old values
      */
     virtual void Predict()
     {
     }
 
     /**
-    Initialization of member variables and prior operations
+     * Initialization of member variables and prior operations
      */
     virtual void Initialize()
     {
+        KRATOS_ERROR << "You are calling to the base class method Initialize, please define in you derived class the method" << std::endl;
     }
 
     /**
-    the problem of interest is solved.
-	This function calls sequentially: Initialize(), InitializeSolutionStep(), Predict(), SolveSolutionStep() and FinalizeSolutionStep().
-	All those functions can otherwise be called separately.
+     * The problem of interest is solved.
+     * This function calls sequentially: Initialize(), InitializeSolutionStep(), Predict(), SolveSolutionStep() and FinalizeSolutionStep().
+     * All those functions can otherwise be called separately.
      */
     virtual double Solve()
     {
-		Initialize();
-		InitializeSolutionStep();
-		Predict();
-		SolveSolutionStep();
-		FinalizeSolutionStep();
-        return 0.00;
+        Initialize();
+        InitializeSolutionStep();
+        Predict();
+        const bool IsConverged = SolveSolutionStep();
```

Comment:
i think we said to leave this as it was...

such feature is used in the linear strategy, so i would do it consistently here